tui/installer/__init__.py

```diff
@@ -72,20 +75,23 @@ def preserve_settings(answers):
     def preserve_timezone(answers):
         if not_preserve_settings(answers):
             return False
-        if 'installation-to-overwrite' not in answers:
-            return False
-        settings = answers['installation-to-overwrite'].readSettings()
-        return 'timezone' in settings and 'request-timezone' not in settings
+        if 'installation-to-overwrite' in answers and \
+                answers['installation-to-overwrite'].settingsAvailable():
+            settings = answers['installation-to-overwrite'].readSettings()
+            return 'timezone' in settings and 'request-timezone' not in settings
+        return False
     not_preserve_timezone = lambda a: not preserve_timezone(a)
 
     def ha_enabled(answers):
         settings = {}
-        if 'installation-to-overwrite' in answers:
+        if 'installation-to-overwrite' in answers and \
+                answers['installation-to-overwrite'].settingsAvailable():
             settings = answers['installation-to-overwrite'].readSettings()
         return 'ha-armed' in settings and settings['ha-armed']
 
     def out_of_order_pool_upgrade_fn(answers):
-        if 'installation-to-overwrite' not in answers:
+        if 'installation-to-overwrite' not in answers and \
+                not answers['installation-to-overwrite'].settingsAvailable()):
```

Comment:
When the `not` is in the individual conditions (language-like), the `and` here needs to be an `or`.

With the `and`, the 2nd condition would be attempted when `'installation-to-overwrite' not in answers` ;-)

I'd also like to mention or discuss formatting in general:

I while ago, I was actively contributing as one of the maintainers of `https://spack.io/` (now I'm just a maintainer on leave, one could say), I was initially not so happy with the team's decision to use the black formatter for all code (and it's a huge Python code base, the complete code was re-formatted using black).

I learnt that while Black being an opinionated code formatter (having no options besides the line length and if string quoting shall be normalized to `"`-strings), it's formatting is actually very readable.

The good thing about black's formatting is, that there the formatting style is clear, there is no ambiguity how black formats it, it's always consistent.

Which is also good in another way: When using black, black formats the code, and the only variable is the maximum line length to use for black formatting.

This allows to not even have to thinking or weighting on how the code should be formatted, as there is one way it will look like after black formatted it.

Now, Of course, we can't reformat all code at once using black, as most larger patch sets would have conflicts at that point, but I'd be in favour of comitting new changes using the formatting from the tool `darker`, which formats only the added lines using black, and using it since many months, I found not even a single problem doing so.

I know, hat the formatting which black/darker provide for multi-line conditions looks as first foreign. But after some time, I've got used to it:

At least to me, it makes the such multi-line if condition more clearer to read, as the
```py
    if (
```
and the closing 
```py
    ):
```
give room to better see what's in the if condition, and the indentation of the two lines is not indented by 8 characters, they are on the same indentation level, which is good:
```suggestion
        if (
            "installation-to-overwrite" not in answers
            or not answers["installation-to-overwrite"].settingsAvailable()
        ):
```
Even if that causes two more lines of code, I think they are worth it to catch such errors better.